assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/index.stories.js

```diff
@@ -72,189 +77,184 @@ const mockEndpoints = ( args ) => {
-		return (
-			<WithTestRegistry registry={ registry } callback={ setupRegistry }>
-				<Story />
-			</WithTestRegistry>
-		);
+		registry
+			.dispatch( MODULES_IDEA_HUB )
+			.receiveError( error, 'updateIdeaState', [ 'Placeholder Idea' ] );
 	},
-];
+};
 
 export const DataUnavailableNew = Template.bind( {} );
 DataUnavailableNew.storyName = 'Data Unavailable: New';
-DataUnavailableNew.decorators = [
-	( Story ) => {
+DataUnavailableNew.args = {
+	setupRegistry: () => {
 		mockEndpoints( { newIdeas: [] } );
-
-		return <Story />;
 	},
-];
+};
 
 export const DataUnavailableSaved = Template.bind( {} );
 DataUnavailableSaved.storyName = 'Data Unavailable: Saved';
-DataUnavailableSaved.decorators = [
-	( Story ) => {
-		mockEndpoints( { savedIdeas: [] } );
-
-		return <Story />;
-	},
-];
 DataUnavailableSaved.args = {
 	defaultActiveTabIndex: 1,
+	setupRegistry: () => {
+		mockEndpoints( { savedIdeas: [] } );
+	},
 };
 
 export const DataUnavailableDrafts = Template.bind( {} );
 DataUnavailableDrafts.storyName = 'Data Unavailable: Drafts';
-DataUnavailableDrafts.decorators = [
-	( Story ) => {
-		mockEndpoints( { draftPostIdeas: [] } );
-
-		return <Story />;
-	},
-];
 DataUnavailableDrafts.args = {
 	defaultActiveTabIndex: 2,
+	setupRegistry: () => {
+		mockEndpoints( { draftPostIdeas: [] } );
+	},
 };
 
 export const DataUnavailableAll = Template.bind( {} );
 DataUnavailableAll.storyName = 'Data Unavailable: All';
-DataUnavailableAll.decorators = [
-	( Story ) => {
+DataUnavailableAll.args = {
+	setupRegistry: () => {
 		mockEndpoints( {
 			draftPostIdeas: [],
 			newIdeas: [],
 			savedIdeas: [],
 		} );
+	},
+};
 
-		return <Story />;
+export const Activity = Template.bind( {} );
+Activity.storyName = 'Activity In Progress';
+Activity.args = {
+	setupRegistry: ( registry ) => {
+		registry
+			.dispatch( MODULES_IDEA_HUB )
+			.setActivity(
+				'ideas/17450692223393508734',
+				IDEA_HUB_ACTIVITY_IS_DELETING
+			);
+		registry
+			.dispatch( MODULES_IDEA_HUB )
+			.setActivity(
+				'ideas/14025103994557865535',
+				IDEA_HUB_ACTIVITY_IS_PINNING
+			);
+		registry
+			.dispatch( MODULES_IDEA_HUB )
+			.setActivity(
+				'ideas/7612031899179595408',
+				IDEA_HUB_ACTIVITY_CREATING_DRAFT
+			);
 	},
-];
+};
+
+export const ActivitiesDone = Template.bind( {} );
+ActivitiesDone.storyName = 'Activities Done';
+ActivitiesDone.args = {
+	setupRegistry: ( registry ) => {
+		registry
+			.dispatch( MODULES_IDEA_HUB )
+			.setActivity(
+				'ideas/17450692223393508734',
+				IDEA_HUB_ACTIVITY_DELETED
+			);
+		registry
+			.dispatch( MODULES_IDEA_HUB )
+			.setActivity(
+				'ideas/14025103994557865535',
+				IDEA_HUB_ACTIVITY_PINNED
+			);
+		registry
+			.dispatch( MODULES_IDEA_HUB )
+			.setActivity(
+				'ideas/7612031899179595408',
+				IDEA_HUB_ACTIVITY_DRAFT_CREATED
+			);
+		registry
+			.dispatch( MODULES_IDEA_HUB )
+			.setActivity(
+				'ideas/2285812891948871921',
+				IDEA_HUB_ACTIVITY_UNPINNED
+			);
```

Comment:
I think we've shyed away from doing this in the past but it's quite common in Gutenberg.
Let's grab the `setActivity` action creator once rather than calling `dispatch` repeatedly. The same could be applied elsewhere here too. 

```suggestion
		const { setActivity } = registry.dispatch( MODULES_IDEA_HUB );
		setActivity(
			'ideas/17450692223393508734',
			IDEA_HUB_ACTIVITY_DELETED
		);
		setActivity(
			'ideas/14025103994557865535',
			IDEA_HUB_ACTIVITY_PINNED
		);
		setActivity(
			'ideas/7612031899179595408',
			IDEA_HUB_ACTIVITY_DRAFT_CREATED
		);
		setActivity(
			'ideas/2285812891948871921',
			IDEA_HUB_ACTIVITY_UNPINNED
		);
```

Reply:
Good catch. Thanks @aaemnnosttv